xl-v1-multipart.go

```diff
@@ -631,9 +631,11 @@ func (xl xlObjects) CompleteMultipartUpload(bucket string, object string, upload
 
 	// Rename if an object already exists to temporary location.
 	uniqueID := getUUID()
-	err = xl.renameObject(bucket, object, minioMetaBucket, path.Join(tmpMetaPrefix, uniqueID))
-	if err != nil {
-		return "", toObjectErr(err, bucket, object)
+	if xl.isObject(bucket, object) {
+		err = xl.renameObject(bucket, object, minioMetaBucket, path.Join(tmpMetaPrefix, uniqueID))
```

Comment:
>Then i think we need to move this check inside renameObject() .
 Is the check needed in every call to `renameObject()`? Another call is made to rename the temporary object to final location, we already know the source is an object in that case so is the check necessary?